helper/resource/testing.go

```diff
@@ -1222,6 +1328,84 @@ func testCheckResourceAttr(is *terraform.InstanceState, name string, key string,
 // when using TestCheckResourceAttrWith and a value is found for the given name and key.
 //
 // When this function returns an error, TestCheckResourceAttrWith will fail the check.
+//
+// Deprecated: State checks have been superseded by ConfigStateChecks.
+// Use the built-int statecheck.ExpectKnownValue state check in combination
+// with a custom knownvalue.Check instead.
+// CheckResourceAttrWithFunc will be removed in the next major version.
+//
+// The following is an example of using statecheck.ExpectKnownValue in combination
+// with a custom knownvalue.Check to replicate the behaviour of TestCheckResourceAttrWith.
```

Comment:
I like the code snippet that references how to use the new `statecheck` package, but for showing an implementation of a custom knownvalue.... do we want to just link the website page for that?

Same for the other custom known value references in deprecation notices.